File: Lampada.java
public class Lampada {
    // Atributos
    String marca;
    String tipo; 
    boolean isRunning;
    int brilho;
    
    // Método construtor
    public Lampada(String marca, String tipo, int brilho) {
        this.marca = marca;
        this.tipo = tipo;
        this.isRunning = false;
        this.brilho = brilho;
    }

    // Ligando a lâmpada
    public void ligar() {
        if (!isRunning) {
            isRunning = true;
            System.out.println("A lâmpada está ligada, brilho: " + brilho + " lúmens");
        } else {
            System.out.println("A lâmpada já está ligada.");
        }
    }

    // Desligando a lâmpada
    public void desligar() {
        if (isRunning) {
            isRunning = false;
            System.out.println("A lâmpada está desligada.");
        } else {
            System.out.println("A lâmpada já está desligada.");
        }
    }

    // Ajustando o brilho
    public void ajustarBrilho(int novoBrilho) {
        if (isRunning) {
            this.brilho = novoBrilho;
            System.out.println("O brilho da lâmpada foi ajustado para " + brilho + " lúmens");
        } else {
            System.out.println("A lâmpada está desligada, precisa ligar primeiro.");
        }
    }

    public static void main(String[] args) {
        Lampada lampada = new Lampada("Philips", "LED", 800);
        System.out.println("A lâmpada foi criada: " + lampada.tipo);

        lampada.ajustarBrilho(600); 
        lampada.ligar();            
        lampada.ajustarBrilho(600); 
        lampada.ajustarBrilho(1000); 
        lampada.desligar();         
        lampada.ajustarBrilho(500); 
    }
}

File: Lampada.py
class Lampada:
    def __init__(self, marca, tipo, brilho):
        self.marca = marca
        self.tipo = tipo
        self.is_running = False
        self.set_brilho(brilho)

    # Ligando a lampada
    def ligar(self):
        if not self.is_running:
            self.is_running = True
            print(f"A lâmpada está ligada, brilho: {self.brilho} lúmens")
        else:
            print("A lâmpada já está ligada.")

    # Desligando a lampada
    def desligar(self):
        if self.is_running:
            self.is_running = False
            print("A lâmpada está desligada.")
        else:
            print("A lâmpada já está desligada.")

    # Ajustando o brilho
    def ajustar_brilho(self, novo_brilho):
        if self.is_running:
            self.set_brilho(novo_brilho)
            print(f"O brilho da lâmpada foi ajustado para {self.brilho} lúmens")
        else:
            print("A lâmpada está desligada, precisa ligar primeiro.")

    def set_brilho(self, brilho):
        if brilho < 0:
            self.brilho = 0
        elif brilho > 1000:
            self.brilho = 1000
        else:
            self.brilho = brilho

    def __str__(self):
        return f"Lampada [marca={self.marca}, tipo={self.tipo}, is_running={self.is_running}, brilho={self.brilho}]"


if __name__ == "__main__":
    lampada = Lampada("Philips", "LED", 800)
    print(lampada)

    lampada.ajustar_brilho(600)
    lampada.ligar()
    lampada.ajustar_brilho(600)
    lampada.ajustar_brilho(1000)
    lampada.desligar()
    lampada.ajustar_brilho(500)
